src/riscv/lib/src/machine_state/page_cache/state.rs

```diff
@@ -35,13 +37,15 @@ const LAST_HALFWORD_PAGE_OFFSET: u64 = PAGE_SIZE
     .checked_sub(std::mem::size_of::<u16>() as u64)
     .expect("page-size must contain at least one halfword");
 
-struct PageEntry<CPE: CodePageEntry> {
+struct PageEntry<CPE: CodePageEntry<MC, M>, MC: MemoryConfig, M: ManagerBase> {
```

Comment:
Are these trait bounds redundant?

```suggestion
struct PageEntry<CPE> {
```